src/rust/engine/src/nodes.rs

```diff
@@ -212,13 +239,12 @@ impl WrappedNode for Select {
             })
             .to_boxed()
         }
-        &tasks::Rule::Intrinsic(Intrinsic { product, input })
-          if product == context.core.types.snapshot
-            && input == context.core.types.directory_digest =>
+        &Rule::Intrinsic(Intrinsic { product, input })
```

Comment:
Same comment as above with abbreviating things -- it can make it more difficult to review. I might actually really appreciate it if you could revert these because I'm finding it a little difficult to read even right now. It's not necessary if you're convinced the improved readability justifies the increase in review difficulty, imho.